h2o-core/src/test/java/water/fvec/TestFrameBuilder.java

```diff
@@ -0,0 +1,245 @@
+    for (int i = (int) start; i < start + length; i++) {
+
+      for (int colIdx = 0; colIdx < vecTypes.length; colIdx++) {
+        switch (vecTypes[colIdx]) {
+          case Vec.T_NUM:
+            nchunks[colIdx].addNum(numericData.get(colIdx)[i]);
+            break;
+          case Vec.T_STR:
+            nchunks[colIdx].addStr(stringData.get(colIdx)[i]);
+            break;
+          default:
+            throw new UnsupportedOperationException("Unsupported Vector type for the builder");
+
+        }
+      }
+    }
+    Frame.closeNewChunks(nchunks);
+  }
+
+  public int getNumCols() {
+    return numCols;
+  }
+
+  public long getNumRows() {
+    return numRows;
+  }
+
+  public String[] getDataForStrCol(int colNum) {
+    return stringData.get(colNum);
+  }
+
+  public double[] getDataForNumCol(int colNum) {
+    return numericData.get(colNum);
+  }
+
+  public long[] getChunkLayout() {
+    return chunkLayout;
+  }
+
+  public String[] getColNames() {
+    return colNames;
+  }
+
+  public byte[] getVecTypes() {
+    return vecTypes;
+  }
+
+  public String getFrameName() {
+    return frameName;
+  }
+
+  /**
+   * Sets the name for the frame. Default name is created if this method is not called.
+   */
+  public TestFrameBuilder withName(String frameName) {
+    this.frameName = frameName;
+    return this;
+  }
+
+  /**
+   * Sets the names for the columns. Default names are created if this method is not called.
+   */
+  public TestFrameBuilder withColNames(String... colNames) {
+    this.colNames = colNames;
+    return this;
+  }
+
+  /**
+   * Sets the vector types.
+   */
+  public TestFrameBuilder withVecTypes(byte... vecTypes) {
+    this.vecTypes = vecTypes;
+    this.numCols = vecTypes.length;
+    return this;
+  }
+
+  /**
+   * Sets data for a particular column
+   *
+   * @param column for which to set data
+   * @param data   array of data
+   */
+  public TestFrameBuilder withDataForCol(int column, String[] data) {
+    stringData.put(column, data);
+    return this;
+  }
+
+  /**
+   * Sets data for a particular column
+   *
+   * @param column for which to set data
+   * @param data   array of data
+   */
+  public TestFrameBuilder withDataForCol(int column, double[] data) {
+    numericData.put(column, data);
+    return this;
+  }
+
+  /**
+   * Sets data for a particular column
+   *
+   * @param column for which to set data
+   * @param data   array of data
+   */
+  public TestFrameBuilder withDataForCol(int column, long[] data) {
+    double[] d = new double[data.length];
+    for (int i = 0; i < data.length; i++) {
+      d[i] = data[i];
+    }
+    numericData.put(column, d);
+    return this;
+  }
+
+  private void checkVecTypes() {
+    assert vecTypes != null && vecTypes.length != 0 : "Vec types has to be specified";
+  }
+
+  public TestFrameBuilder withChunkLayout(long... chunkLayout) {
+    this.chunkLayout = chunkLayout;
+    return this;
+  }
+
```

Comment:
If not too difficult, could you please move the `build()` method here, and the `createChunks()` method anywhere down below, so that the public interface of the class and its private implementation are cleanly separated?